### PR TITLE
Allow service start with default payment price

### PR DESCRIPTION
**Pull request by Waldz**
Fixes: #2116

- Allows optional service price then starting service and if price not given uses YAML configured price, or defaults.
- Price configuration same for all services, no support per service pricing as UI will override them anyways.
- Removed rate notations from Tequilapi service start DTOs, as callers don't understand them
- Price notations in uint64 tokens, to be consistent in Tequilapi scope. Human readable floats left for CLI inputs

### CODE REVIEW

**Comment by soffokl** [comment]
Having a unique price per service was a initial requirement: https://github.com/mysteriumnetwork/node/issues/1962
Why make it common now? 
If UI does not support it, it should be fixed in UI, not reverting in the node.

**Comment by Waldz** [comment]
>> ..as UI will override them anyways.
Persistent config is used by UI only, so they have to be consistent to avoid value conflicts. 
Trying to avoid what we learned from configuration "services to start" being inconsistent with UI forms. And when providers experiencing differences in UI and device restarts.

Per service pricing is achievable in CLI commands by:
`service start 0x1a95ef8027e0dda6b91e7bc8efe68cefc39492a9 wireguard --payment.price-gb=7.5 --payment.price-minute=8.5` 
it it doesn't need to persist passed configuration.

**Comment by soffokl** [comment]
Noone uses service start from CLI, it's ok for development when you starting a service for a few minutes, but not suites for real users. 
How this supposed to be used with auto-updates, Raspberry Pi restarts, docker containers? 

Configuration should be persistent through command-line arguments or configuration files. If it does not survive restart it's useless. 

If it was incompatible with WebUI it supposed to be improved in WebUI, not degraded in the node. 

> Persistent config is used by UI only
That is not true. The configuration file is a configuration file, I'm using it to configure my node.

File: cmd/node.go [hunk 1]
```diff
@@ -1,3 +1,20 @@
+/*
+ * Copyright (C) 2020 The "MysteriumNetwork/node" Authors.
```
Comment (anjmao) [inline cmd/node.go:2]: If you rename file there is no need to update copyright years. Copy right years indicates when the file and it's content was created initially. If somebody pastes this code into it's own library he will not update years too.

File: config/flags_service_noop.go [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ * Copyright (C) 2020 The "MysteriumNetwork/node" Authors.
```
Comment (anjmao) [inline config/flags_service_noop.go:2]: Same here.

File: services/options_service.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
```
Comment (anjmao) [inline services/options_service.go:2]: 2020

File: config/flags_service_shared.go [hunk 4]
```diff
@@ -68,18 +61,25 @@ var (
 		Usage: `Proposal fetch interval { "30s", "3m", "1h20m30s" }`,
 		Value: 10 * time.Minute,
 	}
+
+	// FlagPaymentPricePerGB sets the price per GiB to provided service.
+	FlagPaymentPricePerGB = cli.Float64Flag{
```
Comment (anjmao) [inline config/flags_service_shared.go:66]: Why is this moved to shared if each service should be able to set it's own price?

**Comment by zolia** [comment]
After restart through WEB-UI, consumers fail to connect. 

Provider side log: 
```
2020-04-30T13:45:16.896 DBG p2p/listener.go:209                      > Received consumer public key 770e9e713b1890dd655c0b8bc56a4ed51b7a978d0945963c061641579d70fe67
2020-04-30T13:45:16.896 DBG p2p/listener.go:209                      > Received consumer public key 770e9e713b1890dd655c0b8bc56a4ed51b7a978d0945963c061641579d70fe67
2020-04-30T13:45:17.148 DBG core/ip/resolver.go:103                  > IP detected: 95.216.149.13
2020-04-30T13:45:17.149 INF core/port/pool.go:68                     > Supplying port 48804
2020-04-30T13:45:17.149 INF core/port/pool.go:68                     > Supplying port 48913
2020-04-30T13:45:17.149 DBG p2p/listener.go:244                      > Sending reply with public key 83d6d7aa6a03f7302ada59a36e4192576b0782bb9effb978c63e61434ee8d174 and encrypted config to consumer
2020-04-30T13:45:17.159 DBG core/ip/resolver.go:103                  > IP detected: 95.216.149.13
2020-04-30T13:45:17.160 INF core/port/pool.go:68                     > Supplying port 46271
2020-04-30T13:45:17.160 INF core/port/pool.go:68                     > Supplying port 41458
2020-04-30T13:45:17.160 DBG p2p/listener.go:244                      > Sending reply with public key be90e7d80fd4e3d6d5dff63cbcd60e8dabc5d70683e07d3097910a0103fb9d39 and encrypted config to consumer
2020-04-30T13:45:17.511 ERR p2p/listener.go:124                      > Could not handle exchange ack error="could not decrypt peer conn config: could not decrypt config to proto bytes: could not decrypt message"
2020-04-30T13:45:17.511 ERR p2p/listener.go:124                      > Could not handle exchange ack error="could not decrypt peer conn config: could not decrypt config to proto bytes: could not decrypt message"
```

consumer side log: 
```
2020-04-30T14:45:16.766 INF core/connection/manager.go:624           > Connection state: NotConnected -> Connecting
2020-04-30T14:45:16.766 DBG communication/nats/connector.go:45       > Connecting to NATS servers: nats://testnet-broker.mysterium.network:4222
2020-04-30T14:45:16.767 INF firewall/outgoing_firewall_noop.go:57    > Allow URL nats://testnet-broker.mysterium.network:4222 access
2020-04-30T14:45:16.767 DBG eventbus/event_bus.go:81                 > Published topic="State" event={State:Connecting SessionInfo:{StartedAt:2020-04-30 14:45:16.766497 +0300 EEST m=+47.612191655 ConsumerID:{Address:0x0fc008021a33ee21a6d2c9dbb2a547fd8fc0b84a} AccountantID:[2 20 40 28 241 92 26 102 181 25 144 226 230 94 31 123 124 54 51 24] State:Connecting SessionID: Proposal:{ID:1 Format:service-proposal/v1 ServiceType:openvpn ServiceDefinition:{Location:{Continent:EU Country:DE City:Unknown ASN:24940 ISP:Hetzner Online GmbH NodeType:hosting} LocationOriginate:{Continent:EU Country:DE City:Unknown ASN:24940 ISP:Hetzner Online GmbH NodeType:hosting} SessionBandwidth:8.388608e+07 Protocol:udp} PaymentMethodType:BYTES_TRANSFERRED_WITH_TIME PaymentMethod:{Price:0.000500MYST Duration:600ms Bytes:766958 Type:BYTES_TRANSFERRED_WITH_TIME} ProviderID:0x24a1f81f7410d676cc40b2f212b2b36b9aa951d8 ProviderContacts:[{Type:nats/v1 Definition:{Topic:0x24a1f81f7410d676cc40b2f212b2b36b9aa951d8.openvpn BrokerAddresses:[nats://testnet-broker.mysterium.network:4222]}} {Type:nats/p2p/v1 Definition:{BrokerAddresses:[nats://testnet-broker.mysterium.network:4222]}}] AccessPolicies:<nil>}}}
2020-04-30T14:45:16.767 INF core/state/state.go:359                  > Session ID  Connecting duration: 1.062809ms data: 0b/0b, throughput: 0bs/0bs, spent: 0.000000MYST
2020-04-30T14:45:16.863 DBG p2p/dialer.go:144                        > Consumer 0x0fc008021a33ee21a6d2c9dbb2a547fd8fc0b84a sending public key 770e9e713b1890dd655c0b8bc56a4ed51b7a978d0945963c061641579d70fe67 to provider 0x24a1f81f7410d676cc40b2f212b2b36b9aa951d8
2020-04-30T14:45:17.187 DBG p2p/dialer.go:171                        > Consumer 0x0fc008021a33ee21a6d2c9dbb2a547fd8fc0b84a received provider 0x24a1f81f7410d676cc40b2f212b2b36b9aa951d8 with config: publicIP:"95.216.149.13" ports:48804 ports:48913
2020-04-30T14:45:17.468 DBG core/ip/resolver.go:103                  > IP detected: 78.157.70.161
2020-04-30T14:45:17.469 INF core/port/pool.go:68                     > Supplying port 40459
2020-04-30T14:45:17.469 INF core/port/pool.go:68                     > Supplying port 43912
2020-04-30T14:45:17.469 DBG p2p/dialer.go:194                        > Consumer 0x0fc008021a33ee21a6d2c9dbb2a547fd8fc0b84a sending ack with encrypted config to provider 0x24a1f81f7410d676cc40b2f212b2b36b9aa951d8
```

File: services/options_service.go (hunk 3 above)
Comment (Waldz) [reply services/options_service.go:2]: Not relevant, it's moved file

File: config/flags_service_noop.go (hunk 2 above)
Comment (Waldz) [reply config/flags_service_noop.go:2]: Not relevant, it's new file

File: cmd/node.go (hunk 1 above)
Comment (Waldz) [reply cmd/node.go:2]: Thanks. Here my IDE accidentally added double header. In other noted places Github not some reason wrongly indicates moved files as new ones.

File: services/options_service.go (hunk 3 above)
Comment (Waldz) [reply services/options_service.go:2]: Not relevant, it's moved file

**Comment by Waldz** [comment]
Can't reproduce any of described situations :/

File: config/flags_service_shared.go (hunk 4 above)
Comment (Waldz) [reply config/flags_service_shared.go:51]: Implemented per service parameters for prices and access policies also

File: config/flags_service_start.go [hunk 5]
```diff
@@ -50,60 +39,46 @@ var (
...
-		Value: 10 * time.Minute,
-	}
-	// FlagShaperEnabled enables bandwidth limitation.
-	FlagShaperEnabled = cli.BoolFlag{
-		Name:  "shaper.enabled",
-		Usage: "Limit service bandwidth",
+
+	// FlagPaymentPricePerGB sets the price per GiB to provided service.
+	FlagPaymentPricePerGB = cli.Float64Flag{
+		Name:  "payment.price-gb",
```
Comment (soffokl) [inline config/flags_service_start.go:52]: We have `wireguard.price-gb`, `openvpn.price-gb`, `noop.price-gb`. 
What is `payment.price-gb`?

File: tequilapi/contract/service.go [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package contract
+
+// ServiceStartRequest request used to start a service.
+// swagger:model ServiceStartRequestDTO
```
Comment (anjmao) [inline tequilapi/contract/service.go:21]: Update swagger model to `ServiceStartRequest`

File: config/flags_service_start.go (hunk 5 above)
Comment (anjmao) [reply config/flags_service_start.go:52]: From the code this flag is used as a fallback, for example if `wireguard.price-gb` is not set `payment.price-gb` will be used.

Comment (soffokl) [reply config/flags_service_start.go:52]: But we already have a default value if the flag not set. Do we need another one?

Comment (Waldz) [reply config/flags_service_start.go:52]: As we do not support "per service" configs (except handy yaml configuring).
UI needs single place to store prices same as it was doing with another "per service" config  - access policy to use for service.

Comment (soffokl) [reply config/flags_service_start.go:52]: UI can set 2 places, and it was doing it with no problem.
I'm just worried that we are adding too much confusion for users with these flags overlap.

Comment (Waldz) [reply config/flags_service_start.go:52]: That's why i recommended to continue with no "per service" support for now as we was living with with with access policies.
Storing to 2 places is workaround which leads to broken UI values after manual yaml edits.

File: tequilapi/contract/service.go (hunk 6 above)
Comment (Waldz) [reply tequilapi/contract/service.go:21]: It's not a mistake, i am being consistent with other Swagger models here

File: cmd/commands/service/command.go [hunk 7]
```diff
@@ -101,79 +102,62 @@ func describeQuit(err error) error {
...
-	if err := sc.runServices(ctx, providerID.Address, serviceTypes); err != nil {
-		return err
+		go sc.runService(startRequest)
 	}
 
 	return <-sc.errorChannel
 }
 
-func (sc *serviceCommand) unlockIdentity(identityOptions service.OptionsIdentity) *identity.Identity {
+func (sc *serviceCommand) unlockIdentity(id, passphrase string) string {
```
Comment (soffokl) [inline cmd/commands/service/command.go:143]: What if the passphrase incorrect? It will be locked forever.

File: config/flags_service_openvpn.go [hunk 8]
```diff
@@ -50,13 +50,16 @@ var (
 	FlagOpenVPNPriceMinute = cli.Float64Flag{
 		Name:  "openvpn.price-minute",
 		Usage: "Sets the price of the OpenVPN service per minute.",
-		Value: 0.0001,
```
Comment (soffokl) [inline config/flags_service_openvpn.go:53]: This is how it looks now in the help message:
image

It says the default price for OpenVPN service is 0.
